### DISCUSSION TITLE
Is it possible to get the "Font Bounding Box" within FontForge's Python Modules?

### DISCUSSION THREAD

**Discussion by CubeSolver25** [fontforge/fontforge#4997] [Q&A] [answered]
I've been continuing to work on the SVG exporter, and now there's another problem that I need to address;
**The viewbox of the SVGs are too small to properly display accented glyphs.** 
It's not a new issue for users to struggle with accented glyphs (especially when exporting). Apparently, Fontforge determines the height through `font.ascent + font.descent` which does work for most glyphs.

Using `glyph.boundingBox()` helps somewhat, but because of its wildly different dimensions, I was wondering if there's a better solution than this.
I did find an option that did seem to fix the accented glyph cropping issue, which is to enable **View -> Fit to font bounding box**.

_Without "Fit to font bounding box"_
<img width="712" alt="Screen Shot 2022-04-23 at 10 39 46 AM" src="https://user-images.githubusercontent.com/36614761/164933774-c39ac74d-6ea0-4f24-a591-6d0a3d30e6a4.png">

_With "Fit to font bounding box"_
<img width="712" alt="Screen Shot 2022-04-23 at 10 40 03 AM" src="https://user-images.githubusercontent.com/36614761/164933776-84698aa9-8346-44cf-857e-981c8b44e1d2.png">

After finding this, I tried checking the API to see if `font.boundingBox()` exists, but I couldn't seem to find this function.
Attempting to run such a function also returned an error: 
`AttributeError: 'fontforge.font' object has no attribute 'boundingBox'`
If Fontforge knows a bounding box which can fit all glyphs within the font, how can I get the tuple/values within Python? 

**Alternatively, how does Fontforge determine the font bounding box?**

The only value I need is the **minimum height** that is necessary to fit all glyphs **without** offsetting the position of each glyph by different values to fit inside the viewBox. I want to make sure that if I were to line up these SVGs horizontally, the accented glyphs wouldn't look like they were pushed up or down.

(For the record, I did try `glyph.vwidth`. All glyphs had the same value, even the ones that were cropped.)

**Comment by skef** [comment, marked as answer]
The two fundamental sources of height information are font-wide parameters like ascent and descent, and the individual glyphs. When you're talking about accented characters only the latter will give you what you need. 

Therefore a font-level `boundingBox()` routine would need to iterate over all the glyphs anyway. This would be faster in C but I guess we've never seen a need for it to be a bit faster. So your best bet is to iterate over the glyphs, calling `boundingBox()` on each one and storing the minimum `ymin` and the maximum `ymax`.

**Reply by CubeSolver25** [reply]
Alright, I'll try to see if I can use this method reliably by first iterating through all glyphs found in a font, then exporting based off the collected min `ymin` and max `ymax`. Thank you again!
